sensor/common/detector/enricher.go

```diff
@@ -356,6 +355,16 @@ func (e *enricher) getImages(deployment *storage.Deployment) []*storage.Image {
 	return images
 }
 
+func (e *enricher) getPullSecrets(deployment *storage.Deployment) []string {
+	var pullSecrets []string
+	pullSecrets = append(pullSecrets, deployment.GetImagePullSecrets()...)
+	if len(pullSecrets) == 0 {
+		// Only add service account secrets if the spec did NOT specify image pull secrets.
+		pullSecrets = append(pullSecrets, e.serviceAccountStore.GetImagePullSecrets(deployment.GetNamespace(), deployment.GetServiceAccount())...)
```

Comment:
Not in my testing, also when ACS parses a 'deployment' it will assume the default service account if none found:

https://github.com/stackrox/stackrox/blob/ba8f9376c6c5229126448a45abd9b42cd93d09a7/pkg/protoconv/resources/resources.go#L313-L315